### ISSUE TITLE
SlevomatCodingStandard.TypeHints.TypeHintDeclaration sniff removes annotations about generic parameters

### ISSUE THREAD

**Issue by ste93cry** [slevomat/coding-standard#814]
PHPStan `0.12` started complaining in my project that some parameters typehinted as `ArrayObject` were missing the typehints for both the key and the values as it's now recognized as a generic collection. To solve the issue I updated my code as follow:

```php
class Foo
{
    /**
     * @param ArrayObject<string, mixed> $context
     */
    public function bar(ArrayObject $baz)
    {
        // ...
    }
}
```

However PHP_CodeSniffer with the Slevomat Coding Standard enabled is now complaining that `Method Foo::bar() has useless @param annotation for parameter $baz.`. I know that I can use the vendor-prefixed annotations, but to me this seems a bug so I tracked it down in the class `TypeHintDeclarationSniff`. In particular, the method `getTypeHintFromOneType` is returning just the type name of the param without considering its generic types, so in `getUselessParameterAnnotations` when the `typeHintEqualsAnnotation` is called what happens is that the typehint coming from the PHP source code is considered equal to the annotation without the generic types specified and thus the annotation is marked as useless and removed. To fix the issue it should be enough to change the following lines. If you don't see any cons I can submit a PR

```diff
if ($typeNode instanceof GenericTypeNode) {
-    return $typeNode->type->name;
+    return (string) $typeNode;
}
```

**Comment by kukulich** [comment]
Use the last version please. TypeHintDeclarationSniff does not exist in last version.

**Comment by ste93cry** [comment]
You're right, I feel stupid because I checked this morning to see if I was using the latest version and now I don't know what the heck i saw... Sorry for the invalid bug report
